insights/parsers/lsinitrd.py

```diff
@@ -0,0 +1,91 @@
+"""
+Lsinitrd - command ``lsinitrd``
+===============================
+
+This parser parses the filtered output of command ``lsinitrd`` and provides the
+info of listed files.
+"""
+
+from insights import parser, CommandParser
+from insights.core import ls_parser
+from insights.specs import Specs
+from insights.parsers import keyword_search, SkipException
+
+
+@parser(Specs.lsinitrd)
+class Lsinitrd(CommandParser):
+    """
+    A parser for command "lsinitrd".
+
+    Attributes:
+        data (dict): The key is the filename, the value is a dict describe
+            the file's info.
+        unparsed_lines(list): List of strings for unparsed lines.
+
+    As this lsinitrd spec is set to filterable, the structure of the output
+    is broken. Hence, this parser will parse only filelisting like lines in
+    output of 'lisinitrd', and also store all the unparsed lines.
+    If the other parts of the output structure are required in the future,
+    an enhancement may be performed then.
+
+    Examples:
+        >>> len(ls.data)
+        5
+        >>> assert ls.search(name__contains='kernel') == [
+        ...    {'group': 'root', 'name': 'kernel/x86', 'links': 3, 'perms': 'rwxr-xr-x',
+        ...     'raw_entry': 'drwxr-xr-x   3 root     root            0 Apr 20 15:58 kernel/x86',
+        ...     'owner': 'root', 'date': 'Apr 20 15:58', 'type': 'd', 'dir': '', 'size': 0}
+        ... ]
+        >>> "udev-rules" in ls.unparsed_lines
+        True
+    """
+
+    def parse_content(self, content):
+        file_types = set(['s', 'd', 'p', 'l', '-', 'c', 'b'])
+        perm = set(['-w-', 'rw-', '--x', '-wx', '---', 'rwx', 'r--', 'r-x'])
+        entries = []
+        _unparsed_lines = []
+
+        for l in content:
+            if l and len(l) > 10 and l[0] in file_types and l[1:4] in perm:
+                entries.append(l)
+            else:
+                _unparsed_lines.append(l)
+        self.unparsed_lines = _unparsed_lines
+
+        try:
+            d = ls_parser.parse(entries, '').get('')
+            self.data = d.get('entries')
+        except:
+            raise SkipException("Parsing failure for lsinitrd command output.")
```

Comment:
It's not OK to skip the exception raised by the `parse` since SkipException will just be thrown without any catch, we might miss some unknown formats.

BTW, it's not a good idea to use the empty `''` string the `dir` name, although it's a totally valid key in python.